### DISCUSSION TITLE
Please write a normal readme

### DISCUSSION THREAD

**Discussion by Specialist001** [diadal/universal-social-auth#16] [Show and tell]
I want to use this for Vue 3. But in the "readme" file, everything is mixed up.

**Comment by diadal** [comment]
which part?

**Reply by Specialist001** [reply]
`import axios, { AxiosInstance } from 'axios'
import UniversalSocialauth from 'universal-social-auth'
or import {UniversalSocialauth} from 'universal-social-auth'


`Vue3` declare module '@vue/runtime-core' {
            interface ComponentCustomProperties {
                $axios: AxiosInstance;
                $Oauth:UniversalSocialauth;

            }
            }
const options = {
  providers: {
    apple: {
      nonce: '**************',
      state: '**************',
      clientId: '**************',
      responseMode: 'form_post',
      response_type: 'code  id_token',
      optionalUrlParams: ['response_mode', 'use_popup', 'state'],
      usePopup: true,
      redirectUri: 'https://myapp.com/auth/github/callback'
    },
    github: {
      clientId: '**************',
      redirectUri: 'https://myapp.com/auth/github/callback'
    },
    google: {
      clientId: '***************',
      redirectUri: 'https://myapp.com/auth/google/callback'
    },
    facebook: {
      clientId: '************',
      redirectUri: 'https://myapp.com/auth/facebook/callback'
    },
    twitter: {
      url: 'https://myapp.com/auth/twitter',
      clientId: '********',
      redirectUri: 'https://myapp.com/auth/twitter/callback'
    }
  }
}

const Oauth:UniversalSocialauth = new UniversalSocialauth(axios, options)


`Vue2` Vue.prototype.$axios = axios
`Vue2` Vue.prototype.$Oauth = Oauth

`Vue3` app.config.globalProperties.$Oauth = Oauth
`Vue3` app.config.globalProperties.$axios = axios

`Other Framework` based on your global declaration`

this part.

And this:
`function useAuthProvider (provider:string, proData:Record<string, unknown>| null) {
      const pro = <ProderT>proData

      const ProData = pro || <ProderT>Providers[provider]
      box.$Oauth.authenticate(provider, ProData).then((response) => {
        const rsp:{code:string} = <{code:string}>response
        if (rsp.code) {
          responseData.value.code = rsp.code
          responseData.value.provider = provider
          useSocialLogin()
        }
      }).catch((err:unknown) => {
        console.log(err)
      })
    }`

**Reply by diadal** [reply]
only saw this just now you check the new readme

**Comment by whoacowboy** [comment]
Even though I love this package, I have to agree. The code examples all over the place.

Some issues:

### Vue.js 

```
async function useLoginFirst (e: User) {
    // this sample of how to pust user data to my store
      const firstlogin: boolean = await box.$auth.firstlogin(e)
      if (firstlogin) {
        const apm: string = box.$appName
        $q.notify({
          color: 'positive',
          textColor: 'white',
          message: `Welcome To ${apm}`,
          icon: 'mdi-alarm'
        })
        await box.$router.push({ name: 'dashboard' })
        return
      }
    }

  function useSocialLogin () {
      // otp from input Otp form
      // hash user data in your backend with Cache or save to database
      const pdata = { code: responseData.value.code, otp: data.value.tok, hash: hash.value }
      box.$axios.post('/social-login/' + responseData.value.provider, pdata).then(async (response) => {
          // `response` data base on your backend config
        if (response.data.status === 444) {
          hash.value = response.data.hash
          fauth.value = true // Option show Otp form incase you using 2fa or any addition security apply to your app you can handle all that from here

        }else if (response.data.status === 445) {
          //do something Optional

        }else {

          await useLoginFirst(response.data.u)
        }
      }).catch((err:unknown) => {

        console.log(err)
      })
    }
// Optional for Native App listen to the event `OauthCall` from your page component main.[js|ts] or app.[js|ts]

    emitter.on('OauthCall',  (e) => {
      if(e){

        responseData.value.code = e
        useSocialLogin()

      }


    })
```

Where does `box` come from?

I know that `$q` is from `useQuasar` but someone else might not know.

Then there is `fauth.value` which gets set to true because, yeah.

The php code has some issues as well. 

Numerous references to `clean`, `xlean` & `nlean`. I assume these are some sort of sting cleaning routine but it would be nice to know what it is cleaning and why you put it in there.

Also there is a reference to `Google2FA` which would be great to get more information on this. 

I'd be happy to write up a cleaner version for you but I'd need access to the original project to see what you were trying to do.

On the laravel side I found this Social integration repo that was super helpful.

https://github.com/ankurk91/laravel-socialite-multiple-providers-example

**Reply by diadal** [reply]
to answer you question box is `globalProperties` 

```
const globalProperties = <ComponentInternalInstance>getCurrentInstance();
const box: ComponentCustomProperties = <ComponentCustomProperties>(
  globalProperties.appContext.config.globalProperties
);
```

check the update readme

**Reply by whoacowboy** [reply]
Thanks, the updated readme is better.
